pagerduty/resource_pagerduty_event_rule.go

```diff
@@ -0,0 +1,145 @@
+package pagerduty
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+func resourcePagerDutyEventRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePagerDutyEventRuleCreate,
+		Read:   resourcePagerDutyEventRuleRead,
+		Update: resourcePagerDutyEventRuleUpdate,
+		Delete: resourcePagerDutyEventRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"action_json": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"condition_json": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"advanced_condition_json": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"catch_all": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func buildEventRuleStruct(d *schema.ResourceData) *pagerduty.EventRule {
+	eventRule := &pagerduty.EventRule{
+		Actions:   expandString(d.Get("action_json").(string)),
+		Condition: expandString(d.Get("condition_json").(string)),
+	}
+
+	if attr, ok := d.GetOk("advanced_condition_json"); ok {
+		eventRule.AdvancedCondition = expandString(attr.(string))
+	}
+
+	if attr, ok := d.GetOk("catch_all"); ok {
+		eventRule.CatchAll = attr.(bool)
+	}
+
+	return eventRule
+}
+
+func expandString(v string) []interface{} {
+	var obj []interface{}
+	err := json.Unmarshal([]byte(v), &obj)
+
+	if err != nil {
+		log.Printf(string(err.Error()))
+	}
+
+	return obj
+}
+
+func flattenSlice(v []interface{}) string {
+	b, err := json.Marshal(v)
+	if err != nil {
+		log.Printf(string(err.Error()))
+	}
+	return string(b)
+}
+
+func resourcePagerDutyEventRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*pagerduty.Client)
+
+	eventRule := buildEventRuleStruct(d)
+
+	log.Printf("[INFO] Creating PagerDuty event rule: %s", "eventRule")
+
+	eventRule, _, err := client.EventRules.Create(eventRule)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(eventRule.ID)
+
+	return resourcePagerDutyEventRuleRead(d, meta)
+}
+
+func resourcePagerDutyEventRuleRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*pagerduty.Client)
+
+	log.Printf("[INFO] Reading PagerDuty event rule: %s", d.Id())
+
+	resp, _, err := client.EventRules.List()
+	if err != nil {
+		return err
+	}
+	for _, rule := range resp.EventRules {
+		log.Printf("[DEBUG] Resp rule.ID: %s", rule.ID)
+		if rule.ID == d.Id() {
+			d.Set("action_json", flattenSlice(rule.Actions))
+			d.Set("condition_json", flattenSlice(rule.Condition))
+			d.Set("advanced_condition_json", flattenSlice(rule.AdvancedCondition))
+			d.Set("catch_all", rule.CatchAll)
+		}
+	}
+	// check if eventRule  not  found
+	if _, ok := d.GetOk("action_json"); !ok {
```

Comment:
Does this work? This calls `d.SetId("")` if the response code was a 404. 
I added an example above on how to clear it from the state if the event rule is missing from the list of event rules 🙂

Reply:
It appeared to have worked. 🤷‍♂ But, I've changed that logic to match what you were saying in the above comment.